core/src/main/java/io/ably/lib/util/JsonUtils.java

```diff
@@ -30,7 +31,10 @@ public JsonUtilsObject add(String key, Object value) {
                 json.addProperty(key, (Number) value);
             } else if (value instanceof JsonUtilsObject) {
                 json.add(key, ((JsonUtilsObject) value).toJson());
+            } else {
+                throw new JsonParseException("Unsupported Object Type");
```

Comment:
Keeping in mind [the issue about removing GSON from our public API](https://github.com/ably/ably-java/issues/573), maybe let's not use the `JsonParseException` that's coming from the `gson` library, but instead create our own exception? Or use some other exception that's not coming from a third-party library? :thinking:

Reply:
here it is https://github.com/ably/ably-java/pull/729/commits/1820e44c901e011075d91c7edd249a237f3ea0ab